### PR TITLE
Attempting to make better subscription buttons for TPL and .onion lists

### PR DESCRIPTION
**Pull request by DandelionSprout**
I'd be mightily surprised if I've actually written the code right, but still.

### CODE REVIEW

**Comment by gwarser** [comment]
`props.url.indexOf("http://")`

This function will return index of `http://` substring in `props.url` string. You need to compare it to number. `<`, `>`, `==`.

**Comment by DandelionSprout** [comment]
Hmm, okay. So that's what the `=== -1)` in the original file was for. I guess I can give this one additional try then. Thanks again.

File: src/FilterLists.Web/ClientApp/modules/home/components/linkButtons/SubscribeButton.tsx [hunk 1]
```diff
@@ -10,25 +10,35 @@ interface IProps {
...
         titlePrefix = "";
     }
-
+    
     const hrefTitle = `&amp;title=${encodeURIComponent(props.name)}`;
-    const href = `abp:subscribe?location=${encodeURIComponent(props.url)}${hrefTitle}`;
-    const title =
+    const href =
+          if (props.url.indexOf(".tpl")) {`https://raw.githubusercontent.com/collinbarrett/FilterLists/master/data/TPLSubscriptionAssistant.html`}
+    else {`abp:subscribe?location=${encodeURIComponent(props.url)}${hrefTitle}`};
```
Comment (gwarser) [inline src/FilterLists.Web/ClientApp/modules/home/components/linkButtons/SubscribeButton.tsx]: Ugly:
```js
const href = props.url.indexOf(".tpl") > 0 ? `https://raw.githubusercontent.com/collinbarrett/FilterLists/master/data/TPLSubscriptionAssistant.html` : `abp:subscribe?location=${encodeURIComponent(props.url)}${hrefTitle}`;
```

`.indexOf(".tpl") > 0` may be `.endsWith(".tpl")`

With `let`:

```js
let href;
if (props.url.indexOf(".tpl") > 0) 
{
    href = `https://raw.githubusercontent.com/collinbarrett/FilterLists/master/data/TPLSubscriptionAssistant.html`;
} else {
    href = `abp:subscribe?location=${encodeURIComponent(props.url)}${hrefTitle}`;
};
```

**Comment by DandelionSprout** [comment]
My coding skills are pretty poor, so I'll have a hard time just by trying to copy over the suggestions from the code reviews, but I'll try.

**Comment by DandelionSprout** [comment]
This may or may not sound amateurish of me to ask, but I seemingly can't even succeed when being guided on this. Would you want to post an entire redo of that file as a comment (or alternately as an all-new pull request)?

**Comment by DandelionSprout** [comment]
Nevermind, apparently the build somehow passed! 🎉 

You have my greatest gratitudes for this, @gwarser, and I'm pretty confident that you'll have Collin's gratitudes as well.

**Comment by collinbarrett** [comment]
Thanks all, were we able to test the effects of this locally? I'm pretty swamped right now, so I'm going to go ahead and pull it in and we'll just test on the live site like real hackers. If the change fails, we can do a new PR to fix or revert.

**Comment by DandelionSprout** [comment]
>were we able to test the effects of this locally?

I have no idea how to test such things locally, so I just went by the Travis scans. 😅

**Comment by DandelionSprout** [comment]
The TPL href change seems to have taken place already, and it went splendidly.

The .onion test hinges on the site regenerating its list of lists, which in the past few weeks have only happened roughly every 40 hours on average... 😥

**Comment by DandelionSprout** [comment]
Update:

The changes made to the JSON files are now live, making me able to prove that the Onion button works as well.
image

**Pull request merged**